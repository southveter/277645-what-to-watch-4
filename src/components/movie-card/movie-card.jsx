import React, {PureComponent} from "react";
import PropTypes from "prop-types";

export class MovieCard extends PureComponent {

  render() {
    const {movieInfo, onClick, onHover} = this.props;
    const {title, poster} = movieInfo;

    return (
      <article
        className="small-movie-card catalog__movies-card">
        <div className="small-movie-card__image"
          onMouseOver={() => {
            onHover(movieInfo);
          }}>
          <img src={poster} alt={title} width="280" height="175" onClick={onClick}/>
        </div>
        <h3 className="small-movie-card__title" onClick={onClick}>
          <a className="small-movie-card__link" href="movie-page.html">{title}</a>
        </h3>
      </article>
    );
  }
}

MovieCard.propTypes = {
  onHover: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  movieInfo: PropTypes.shape({
    title: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
  }).isRequired,
};
