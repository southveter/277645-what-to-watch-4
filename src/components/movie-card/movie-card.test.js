import React from "react";
import renderer from "react-test-renderer";
import {MovieCard} from './movie-card.jsx';


const mocks = {
  title: `The Big Lebowski`,
  poster: `img/what-we-do-in-the-shadows.jpg`,
};
const onClick = () => {};
const onHover = () => {};

it(`Should render correctly`, () => {
  const tree = renderer
    .create(
        <MovieCard
          movieInfo={mocks}
          onClick={onClick}
          onHover={onHover}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

