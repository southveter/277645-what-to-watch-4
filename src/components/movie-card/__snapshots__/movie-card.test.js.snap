// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
    onMouseOver={[Function]}
  >
    <img
      alt="The Big Lebowski"
      height="175"
      onClick={[Function]}
      src="img/what-we-do-in-the-shadows.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      The Big Lebowski
    </a>
  </h3>
</article>
`;
